src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,140 @@
+(ns quo2.components.community.token-gating
+  (:require [quo.design-system.colors :as colors]
+            [quo.react-native :as rn]
+            [quo2.components.avatars.channel-avatar :as channel-avatar]
+            [quo2.components.buttons.button :as button]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.tags.token-tag :as token-tag]
+            [quo2.foundations.colors :as quo2-colors]))
+
+(def styles {:container {:flex 1 :background-color (:ui-background @colors/theme)}
+             :header-container {:flex-direction :row
+
+                                :align-items :center}
+             :header-spacing-community {:margin-bottom 16}
+             :header-spacing-channel {:margin-bottom 24}
+             :header-avatar {:margin-right 12}
+             :header-community-avatar-image {:width 32 :height 32}
+             :header-text {:flex 1}
+             :header-info-button {:margin-left 8}
+             :token-requirement-list-spacing {:padding-vertical 8}
+             :token-list {:align-items :center}
+             :token-row {:flex-direction :row}
+             :token-tag-spacing {:margin-left 7}
+             :divider {:height 1
+                       :background-color quo2-colors/neutral-10
+                       :margin-horizontal -20
+                       :margin-top 12
+                       :margin-bottom 8}
+             :enter-button {:margin-top 16}
+             :info-text {:margin-top 12
+                         :color quo2-colors/neutral-50
+                         :text-align :center
+                         :padding-horizontal 20}})
+
+(defn is-token-requirement-met? [token-requirement-list] (reduce
+                                                          #(or (and %1 (:is-sufficient %2)) false)
+                                                          true
+                                                          token-requirement-list))
+
+(defn token-requirement-list [props]
+  (let [{:keys [gate]} props
+        [gate-type token-requirement-list] gate
+        is-sufficient? (is-token-requirement-met? token-requirement-list)
+        message (if
+                 (= gate-type :join)
+                  (str "Join requirements" (when-not is-sufficient? " not") " met")
```

Comment:
I think these strings should be wrapped in i18n labels and the string to the en.json files

Reply:
Thanks!